internal/outpost/proxyv2/filesystemstore/filesystemstore.go

```diff
@@ -0,0 +1,203 @@
+package filesystemstore
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/gorilla/sessions"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	SessionCleanupInterval     = 5 * time.Minute
+	SessionCleanupLockFileName = "session-cleanup.lock"
+	SessionFilePrefix          = "session_"
+)
+
+var (
+	ErrSessionCleanupAlreadyRunning = errors.New("session cleanup is already running by another instance")
+)
+
+type Store struct {
+	sessions.FilesystemStore
+	storePath string
+	log       *log.Entry
+}
+
+func NewStore(path string, keyPairs ...[]byte) *Store {
+	if path == "" {
+		path = os.TempDir()
+	}
+	return &Store{
+		FilesystemStore: *sessions.NewFilesystemStore(path, keyPairs...),
+		storePath:       path,
+		log:             log.WithField("logger", "authentik.outpost.proxyv2.filesystemstore"),
+	}
+}
+
+// SessionCleanup acquires a file lock to ensure only one instance runs at a time,
+// then checks and deletes expired session files from the filesystem session store.
+// It supports context-based cancellation to allow graceful shutdowns or timeouts.
+func (s *Store) SessionCleanup(ctx context.Context) error {
+	s.log.Info("Starting session cleanup")
+	lockPath := path.Join(s.storePath, SessionCleanupLockFileName)
+	lockFile, err := os.OpenFile(lockPath, os.O_CREATE|os.O_RDWR, 0644)
```

Comment:
```suggestion
	lockFile, err := os.OpenFile(lockPath, os.O_CREATE|os.O_RDWR, 0600)
```

would it work with this permission?

Reply:
Yes, also works with `0600`. Is modified 😃